src/chrome/cdtpDebuggee/features/cdtpBrowserNavigator.ts

```diff
@@ -0,0 +1,39 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+import { Protocol as CDTP } from 'devtools-protocol';
+
+import { injectable, inject } from 'inversify';
+import { TYPES } from '../../dependencyInjection.ts/types';
+import { CDTPEventsEmitterDiagnosticsModule } from '../infrastructure/cdtpDiagnosticsModule';
+import { CDTPDomainsEnabler } from '../infrastructure/cdtpDomainsEnabler';
+
+export interface IBrowserNavigator {
+    navigate(params: CDTP.Page.NavigateRequest): Promise<CDTP.Page.NavigateResponse>;
+    reload(params: CDTP.Page.ReloadRequest): Promise<void>;
+    onFrameNavigated(listener: (params: CDTP.Page.FrameNavigatedEvent) => void): void;
+}
+
+@injectable()
+export class CDTPBrowserNavigator extends CDTPEventsEmitterDiagnosticsModule<CDTP.PageApi> implements IBrowserNavigator {
+    protected api = this._protocolApi.Page;
+
+    public readonly onFrameNavigated = this.addApiListener('frameNavigated', (params: CDTP.Page.FrameNavigatedEvent) => params);
```

Comment:
Looks like this is supposed to return void

Reply:
I don't understand what you mean.

Reply:
I was looking at the types of `IBrowserNavigator#onFrameNavigated` but I just read it wrong, nvm.